### PR TITLE
Send client sample rate as a span measurement

### PR DESCRIPTION
**Pull request by colin-sentry**
This will be consumed by snuba spans consumer

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 **Internal**:
 
 - Add `EnvelopeStack` and `SQLiteEnvelopeStack` to manage envelopes on disk. ([#3855](https://github.com/getsentry/relay/pull/3855))
+- Add `client_sample_rate` to spans, pulled from the trace context ([#3872](https://github.com/getsentry/relay/pull/3872))
```
Comment (jjbayer) [inline CHANGELOG.md]:
```suggestion
- Add `client_sample_rate` to spans, pulled from the trace context ([#3872](https://github.com/getsentry/relay/pull/3872)).
```

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 2]
```diff
@@ -304,6 +306,12 @@ fn extract_shared_tags(event: &Event) -> BTreeMap<SpanTagKey, String> {
         if let Some(status) = trace_context.status.value() {
             tags.insert(SpanTagKey::TraceStatus, status.to_string());
         }
+
+        if let Some(client_sample_rate) = trace_context.client_sample_rate.value() {
+            if *client_sample_rate > 0. {
+                tags.insert(SpanTagKey::ClientSampleRate, client_sample_rate.to_string());
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]: Should we insert the sample rate as a `span.measurement` instead? That way, it's sent as a number and won't be indexed unnecessarily.

Comment (colin-sentry) [reply relay-event-normalization/src/normalize/span/tag_extraction.rs]: sure, where would that logic go? Measurement extraction doesn't have access to the context

Comment (jjbayer) [reply relay-event-normalization/src/normalize/span/tag_extraction.rs]: If you add it in `extract_segment_measurements` it should end up on the span automatically.

File: relay-event-normalization/src/normalize/span/tag_extraction.rs [hunk 3]
```diff
@@ -794,6 +806,27 @@ fn value_to_f64(val: Option<&Value>) -> Option<f64> {
...
+    let mut measurements = Measurements::default();
+
+    if let Some(trace_context) = event.context::<TraceContext>() {
+        if let Some(client_sample_rate) = trace_context.client_sample_rate.value() {
+            if *client_sample_rate > 0. {
+                measurements.insert(
+                    "client_sample_rate".into(),
+                    Measurement {
+                        value: (*client_sample_rate).into(),
+                        unit: MetricUnit::None.into(),
```
Comment (jjbayer) [inline relay-event-normalization/src/normalize/span/tag_extraction.rs]: nit

```suggestion
                        unit: MetricUnit::Fraction(FractionUnit::Ratio).into(),
```

**Review by jjbayer — APPROVED** [review]

**Pull request merged**